src/runtime_src/core/include/xrt/detail/ert.h

```diff
@@ -589,6 +589,36 @@ struct ert_access_valid_cmd {
   uint32_t wr_test;
 };
 
+/**
+ * struct ert_ctx_health_data: interpretation of payload for an ert packet
+ *                             which has context health data
+ *
+ * @version:          context health data version (current 0.0)
+ * @txn_op_idx:       index of last TXN control code executed
+ * @ctx_pc:           program counter for that context
+ * @fatal_error_type: the fatal error type if context crashes
+ * @aie_data_size:    size in bytes of the aie_data
+ * @aie_data:         binary blob dumpped from aie partition
+ *
+ * Field              Default value  Comment
+ * txn_op_idx:        0xFFFFFFFF     there is no txn control code is running or the
+ *                                   last txn control code op idx is not captured
+ * ctx_pc:            0              context .text program counter is not captured
+ * fatal_error_type:  0              no fatal error or fatal error is not captured
+ * aie_data_size:     0              aie register data is not captured
+ *
+ * Once an ert packet completes with state ERT_CMD_STATE_TIMEOUT, the ert
+ * packet starting from payload will have the following information.
+ */
+struct ert_ctx_health_data {
```

Comment:
One comment.   Binary data like this is hard to keep ABI compatible.  Would it be worth while to reserve some number of words prior to `aie_data[]` ?

Reply:
Larry says we don't need this because we have ctx_healt.version which can be used to ensure proper implementation of any accessors.